### PR TITLE
NEW PROVIDER: Gcore DNS

### PR DESCRIPTION
**Pull request by xddxdd**
This PR adds support for Gcore DNS (https://gcore.com/dns) using their [Golang SDK](https://github.com/G-Core/gcore-dns-sdk-go).

Code is based on deSEC provider, since Gcore performs incremental-label-type updates.

Passes all integration tests, and tested on my own domain.

### CODE REVIEW

File: providers/gcore/convert.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	for _, value := range n.Records {
+		rc := &models.RecordConfig{
+			TTL:      uint32(n.TTL),
+			Original: n,
+		}
+		rc.SetLabelFromFQDN(recName, zoneName)
+		switch recType {
+		case "CAA": // G-Core API don't need quotes around CAA with whitespace
+			if len(value.Content) != 3 {
+				panic("incorrect number of fields in G-Core's CAA record")
```
Comment (tlimoncelli) [inline providers/gcore/convert.go:25]: Please return an error instead of calling panic().

File: providers/gcore/convert.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+			}
+
+			parts := make([]string, len(value.Content))
+			for i := range value.Content {
+				parts[i] = fmt.Sprint(value.Content[i])
+			}
+
+			flag, tag, target := parts[0], parts[1], parts[2]
+			if err := rc.SetTargetCAAStrings(flag, tag, target); err != nil {
+				panic(fmt.Errorf("unparsable record received from G-Core: %w", err))
```
Comment (tlimoncelli) [inline providers/gcore/convert.go:35]: Please return an error instead of calling panic().

File: providers/gcore/convert.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+			}
+
+		default: //  "A", "AAAA", "CAA", "NS", "CNAME", "MX", "PTR", "SRV", "TXT"
+			if err := rc.PopulateFromString(recType, value.ContentToString(), zoneName); err != nil {
+				panic(fmt.Errorf("unparsable record received from G-Core: %w", err))
```
Comment (tlimoncelli) [inline providers/gcore/convert.go:40]: Please return an error instead of calling panic().

**Review by tlimoncelli — COMMENTED** [review]
Excellent work!  Just 1 change (repeated 3 times).

**Comment by xddxdd** [comment]
Thanks for your feedback! That has been fixed.

**Comment by tlimoncelli** [comment]
Excellent!

Thanks for the contribution!  It's always a pleasure to get a new provider added to the project!

**Pull request merged**